src/org/python/core/PyMemoryView.java

```diff
@@ -67,8 +67,30 @@ static PyObject memoryview_new(PyNewWrapper new_, boolean init, PyType subtype,
         ArgParser ap = new ArgParser("memoryview", args, keywords, "object");
         PyObject obj = ap.getPyObject(0);
 
-        if (obj instanceof BufferProtocol) {
-            return new PyMemoryView((BufferProtocol)obj);
+
```

Comment:
I would say the comment is not needed, except to say that `PyUnicode` doesn't really support the buffer protocol. The PyArray bit will out of scope. We would only give background (e.g. about pip and msgpack) to explain a _non_-standard or unexpected choice. As you've pointed out, our motive here is conformance to expectations.

Also, it was too wide. You may find the [coding guidance](https://wiki.python.org/jython/JythonDeveloperGuide#Coding_guidance) useful.